internal/backend/discussion.go

```diff
@@ -258,9 +258,95 @@ func (d *delphisBackend) UpdateDiscussion(ctx context.Context, id string, input
 
 	updateDiscussionObj(discObj, input)
 
+	d.CreateDiscussionArchive(ctx, id)
```

Comment:
It looks like you're creating a discussion archive here and then again below -- is this on purpose?